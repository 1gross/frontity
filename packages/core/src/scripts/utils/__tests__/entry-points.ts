import fsExtra from "fs-extra";
import {
  generateClientEntryPoints,
  generateServerEntryPoint
} from "../entry-points";

jest.mock("fs-extra");
const mockedFsExtra = fsExtra as jest.Mocked<typeof fsExtra>;

const site = [
  {
    name: "site-1",
    mode: "html",
    packages: ["package1", "package2"]
  }
];
const sites = [
  {
    name: "site-html",
    mode: "html",
    packages: ["package1", "package2", "package2"]
  },
  {
    name: "site-amp",
    mode: "amp",
    packages: ["package1", "package3"]
  },
  {
    name: "site-2",
    mode: "html",
    packages: ["package1"]
  }
];

beforeEach(() => {
  mockedFsExtra.writeFile.mockReset();
});

test("it should write one client entry point", async () => {
  await generateClientEntryPoints({ sites: site, outDir: "/build" });
  expect({
    in: site,
    out: mockedFsExtra.writeFile.mock.calls[0]
  }).toMatchSnapshot();
});

test("it should write multiple client entry points", async () => {
  await generateClientEntryPoints({ sites, outDir: "/build" });
  expect({
    in: sites,
    out: mockedFsExtra.writeFile.mock.calls
  }).toMatchSnapshot();
});

test("it should write one server entry point for one site", async () => {
  await generateServerEntryPoint({ sites: site, outDir: "/build" });
  expect({
    in: site,
    out: mockedFsExtra.writeFile.mock.calls[0]
  }).toMatchSnapshot();
});

test("it should write one server entry point for multiple sites", async () => {
  await generateServerEntryPoint({ sites, outDir: "/build" });
  expect({
    in: sites,
    out: mockedFsExtra.writeFile.mock.calls
  }).toMatchSnapshot();
});
