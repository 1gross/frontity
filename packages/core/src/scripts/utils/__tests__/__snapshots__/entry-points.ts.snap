// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should write multiple client entry points 1`] = `
Object {
  "in": Array [
    Object {
      "mode": "html",
      "name": "site-html",
      "packages": Array [
        "package1",
        "package2",
        "package2",
      ],
    },
    Object {
      "mode": "amp",
      "name": "site-amp",
      "packages": Array [
        "package1",
        "package3",
      ],
    },
    Object {
      "mode": "html",
      "name": "site-2",
      "packages": Array [
        "package1",
      ],
    },
  ],
  "out": Array [
    Array [
      "/build/bundling/entry-points/site-html.js",
      "import * as package1 from \\"package1/src/html/client\\";
import * as package2 from \\"package2/src/html/client\\";

export {
  package1,
  package2,
};",
      "utf8",
    ],
    Array [
      "/build/bundling/entry-points/site-amp.js",
      "import * as package1 from \\"package1/src/amp/client\\";
import * as package3 from \\"package3/src/amp/client\\";

export {
  package1,
  package3,
};",
      "utf8",
    ],
    Array [
      "/build/bundling/entry-points/site-2.js",
      "import * as package1 from \\"package1/src/html/client\\";

export {
  package1,
};",
      "utf8",
    ],
  ],
}
`;

exports[`it should write one client entry point 1`] = `
Object {
  "in": Array [
    Object {
      "mode": "html",
      "name": "site-1",
      "packages": Array [
        "package1",
        "package2",
      ],
    },
  ],
  "out": Array [
    "/build/bundling/entry-points/site-1.js",
    "import * as package1 from \\"package1/src/html/client\\";
import * as package2 from \\"package2/src/html/client\\";

export {
  package1,
  package2,
};",
    "utf8",
  ],
}
`;

exports[`it should write one server entry point for multiple sites 1`] = `
Object {
  "in": Array [
    Object {
      "mode": "html",
      "name": "site-html",
      "packages": Array [
        "package1",
        "package2",
        "package2",
      ],
    },
    Object {
      "mode": "amp",
      "name": "site-amp",
      "packages": Array [
        "package1",
        "package3",
      ],
    },
    Object {
      "mode": "html",
      "name": "site-2",
      "packages": Array [
        "package1",
      ],
    },
  ],
  "out": Array [
    Array [
      "/build/bundling/entry-points/server.js",
      "import * as package1 from \\"package1/src/html/server\\";
import * as package2 from \\"package2/src/html/server\\";
import * as package1 from \\"package1/src/amp/server\\";
import * as package3 from \\"package3/src/amp/server\\";

export {
  package1,
  package2,
  package1,
  package3,
};",
      "utf8",
    ],
  ],
}
`;

exports[`it should write one server entry point for one site 1`] = `
Object {
  "in": Array [
    Object {
      "mode": "html",
      "name": "site-1",
      "packages": Array [
        "package1",
        "package2",
      ],
    },
  ],
  "out": Array [
    "/build/bundling/entry-points/server.js",
    "import * as package1 from \\"package1/src/html/server\\";
import * as package2 from \\"package2/src/html/server\\";

export {
  package1,
  package2,
};",
    "utf8",
  ],
}
`;
